### PR TITLE
chore(dependencies): Update superagent version

### PR DESCRIPTION
**Pull request by Tetsu9901**
We want to update the superagent version

### CODE REVIEW

File: src/superagent-mock.js [hunk 1]
```diff
@@ -93,21 +116,6 @@ module.exports = function (superagent, config, logger) {
...
-        this.path = this.url;
-        this._appendQueryString(this); // use superagent implementation of adding the query
-        path = this.path; // save the url together with the query
-        this.path = originalPath; // reverse the addition of query to path by _appendQueryString
-      } else { // client
-        const originalUrl = this.url;
-        this._appendQueryString(this); // use superagent implementation of adding the query
-        path = this.url; // save the url together with the query
-        this.url = originalUrl; // reverse the addition of query to url by _appendQueryString
-      }
```
Comment (Tetsu9901) [inline src/superagent-mock.js:110]: I change only this in this file

File: .eslintrc [hunk 2]
```diff
@@ -7,7 +7,6 @@
     "no-underscore-dangle": 0,
     "no-use-before-define": [2, "nofunc"],
     "max-len": [2, 150, 2],
-    "quotes": [2, "single"],
```
Comment (fdubost) [inline .eslintrc:10]: Why this change? 👀 It generates a lot of noise in this PR 😕

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -10,8 +10,8 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [10.x, 12.x, 14.x, 15.x]
```
Comment (fdubost) [inline .github/workflows/node.js.yml:13]: We should maybe keep the compatibility with node 16 because it's a LTS 🦒

File: package.json [hunk 4]
```diff
@@ -16,10 +16,10 @@
     "url": "https://github.com/M6Web/superagent-mock"
   },
   "engines": {
-    "node": ">=10"
+    "node": "^18.0.0"
```
Comment (fdubost) [inline package.json:19]: As said above, maybe 16 is better

File: .eslintrc (hunk 2 above)
Comment (Tetsu9901) [inline .eslintrc:10]: My prettier modifie all files with `" "` instead of `' '` and
I can't do this : 

``` 
"quotes": [2, "single", "double"]
```
So I removed it instead of revert all prettier modifications

Comment (Tetsu9901) [reply .eslintrc:10]: My prettier modifie all files with " " instead of ' ' and
I can't do this :

"quotes": [2, "single", "double"]
So I removed it instead of revert all prettier modifications

Comment (fdubost) [reply .eslintrc:10]: I just saw your comment above but it is not a good reason to change lint rules in this project 😄 You should find why your prettier is misconfigured. Maybe a `.prettierrc` is missing here.

Comment (Tetsu9901) [reply .eslintrc:10]: Yes it was planned but you are too fast for review 😅

File: .prettierrc [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+    "singleQuote": true,
+    "printWidth": 120,
+    "trailingComma": "all",
+    "bracketSpacing": false
+}
```
Comment (Tetsu9901) [inline .prettierrc:6]: There rules come from app-bedrock-web

File: package.json [hunk 6]
```diff
@@ -16,10 +16,10 @@
     "url": "https://github.com/M6Web/superagent-mock"
   },
   "engines": {
-    "node": ">=10"
+    "node": ">=16.0.0"
   },
   "peerDependencies": {
-    "superagent": ">=3.6.0"
+    "superagent": ">=8.0.9"
```
Comment (SimonChaumet) [inline package.json:22]: Why is superagent a peer dependency instead of direct dependency ?

**Review by Art-G — APPROVED** [review]

File: .prettierrc (hunk 5 above)
Comment (SimonChaumet) [reply .prettierrc:6]: It is a best practice to separate the refacto of prettier and the new code you want to add, here we can't see if a modification comes from the prettier refacto or not

File: yarn.lock
Comment (SimonChaumet) [inline yarn.lock:1]: Are we not supposed to move all projects to pnpm ?

File: tests/support/expectations.js [hunk 7]
```diff
@@ -10,7 +10,6 @@ module.exports = function (request, config, isServer) {
     currentLog = log;
   };
   var superagentPackage = require('superagent/package.json');
-  var superagentUserAgentHeader = isServer ? {'User-Agent': 'node-superagent/' + superagentPackage.version} : {};
```
Comment (SimonChaumet) [inline tests/support/expectations.js:13]: Why does this have been removed ?

**Review by SimonChaumet — APPROVED** [review]

File: package.json (hunk 6 above)
Comment (fdubost) [reply package.json:22]: `superagent-mock` is not in responsibility for installing `superagent` in the project where it is used

File: tests/support/expectations.js (hunk 7 above)
Comment (Tetsu9901) [reply tests/support/expectations.js:13]: The user agent is no longer set by the superagent library, so there's no point in testing its presence on the superagent-mock side
